resources/aws_ec2.go

```diff
@@ -785,6 +785,13 @@ func (obj *AwsEc2Res) CheckApply(apply bool) (checkOK bool, err error) {
 	ctx, cancel := context.WithTimeout(context.TODO(), waitTimeout*time.Second)
 	defer cancel()
 	if len(diOutput.Reservations) == 1 {
+		// If the instance is in the process of starting up or stopping
+		// we cannot affect its state until it finishes and Watch sends
+		// an event.
+		initialState := *diOutput.Reservations[0].Instances[0].State.Name
```

Comment:
Is it possible that it gets into this state *right* after we check this?

Reply:
@purpleidea the short answer is yes. 
Instead I should be checking for the error that's produced where I try to start/stop/terminate the instances and return false, nil there.